### DISCUSSION TITLE
Error (6275): Source "ds-test/test.sol" not found: File not found when running "forge test" // FundMe Test

### DISCUSSION THREAD

**Discussion by ShawnFromTexas** [Cyfrin/foundry-full-course-cu#1281] [Q&A]
i am getting an  this error while running forge test command
error is

```solidity
[⠊] Compiling...
[⠒] Unable to resolve imports:
      "ds-test/test.sol" in "/root/hh-fcc/ethers-simple-storage/foundry-simple-storage-f23/foundry-fund-me-f23/lib/forge-std/src/StdAssertions.sol"
      "ds-test/test.sol" in "/root/hh-fcc/ethers-simple-storage/foundry-simple-storage-f23/foundry-fund-me-f23/lib/forge-std/src/Test.sol"
with remappings:
      @chainlink/contracts/=/root/hh-fcc/ethers-simple-storage/foundry-simple-storage-f23/foundry-fund-me-f23/lib/chainlink-brownie-contracts/contracts/
      chainlink-brownie-contracts/=/root/hh-fcc/ethers-simple-storage/foundry-simple-storage-f23/foundry-fund-me-f23/lib/chainlink-brownie-contracts/contracts/src/v0.6/vendor/@arbitrum/nitro-contracts/src/
      forge-std/=/root/hh-fcc/ethers-simple-storage/foundry-simple-storag[⠢] Unable to resolve imports:forge-std/src/
      "ds-test/test.sol" in "/root/hh-fcc/ethers-simple-storage/foundry-simple-storage-f23/foundry-fund-me-f23/lib/forge-std/src/StdAssertions.sol"
      "ds-test/test.sol" in "/root/hh-fcc/ethers-simple-storage/foundry-simple-storage-f23/foundry-fund-me-f23/lib/forge-std/src/Test.sol"
with remappings:
      @chainlink/contracts/=/root/hh-fcc/ethers-simple-storage/foundry-simple-storage-f23/foundry-fund-me-f23/lib/chainlink-brownie-contracts/contracts/
      chainlink-brownie-contracts/=/root/hh-fcc/ethers-simple-storage/foundry-simple-storage-f23/foundry-fund-me-f23/lib/chainlink-brownie-contracts/contracts/src/v0.6/vendor/@arbitrum/nitro-contracts/src/
      forge-std/=/root/hh-fcc/ethers-simple-storage/foundry-simple-storage-f23/foundry-fund-me-f23/lib/forge-std/src/
[⠰] Compiling 23 files with 0.8.19
[⠔] Solc 0.8.19 finished in 126.47ms
Error: 
Compiler run failed:
Error (6275): Source "ds-test/test.sol" not found: File not found. Searched the following locations: "/root/hh-fcc/ethers-simple-storage/foundry-simple-storage-f23/foundry-fund-me-f23".
 --> lib/forge-std/src/StdAssertions.sol:4:1:
  |
4 | import {DSTest} from "ds-test/test.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error (6275): Source "ds-test/test.sol" not found: File not found. Searched the following locations: "/root/hh-fcc/ethers-simple-storage/foundry-simple-storage-f23/foundry-fund-me-f23".
  --> lib/forge-std/src/Test.sol:27:1:
   |
27 | import {DSTest} from "ds-test/test.sol";
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Here is my code in FundMe, but I believe its because dsTest folders are empty.?

```solidity
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.19;

import {Test, console} from "forge-std/Test.sol";

contract FundMeTest is Test {
    uint256 number = 1;

    function setUp() external {
        number = 2;
    }

    function testDemo() public {
        console.log(number);
        console.log("Hello!");
        assertEq(number, 2);
    }
}
```

Any help is MUCH appreciated!

**Comment by ShawnFromTexas** [comment]
When i hover over 

```solidity
import {DSTest} from "ds-test/test.sol";
```

i get this message. Like no file is there??

```solidity
Source "ds-test/test.sol" not found: File not found. Searched the following locations: "".solidity(6275)
```

**Comment by xyz899** [comment]
Hi,

try this : `forge install foundry-rs/forge-std` , restart vs code or your machine and see if it stills give you the same error

**Reply by xyz899** [reply]
### I just had the same error this morning solved it by using my instructions here

**Reply by ShawnFromTexas** [reply]
Yes mate! Thank you i figured it out LATEEE last night and never updated Github. Appreciate you and have a blessed day winter!

**Reply by xyz899** [reply]
> Yes mate! Thank you i figured it out LATEEE last night and never updated Github. Appreciate you and have a blessed day winter!

Thank you friend and I'm Shawn as well so we buddies now

**Reply by ShawnFromTexas** [reply]
OK;LG

**Comment by ShawnFromTexas** [comment]
I needed to install forge-std again. then re-run VScode again. Had to clean up a file i messed up trying to repair issue earlier as well. oofff

```solidity
forge install foundry-rs/forge-std@v1.5.3 --no-commit
```

this is what i used just in case the above answer from winter doesnt work for the next enjooooyer. cheers!

**Reply by DCVglobalnetwork** [reply]
![Screenshot 2024-01-19 114116](https://github.com/Cyfrin/foundry-full-course-f23/assets/105791829/8601d17e-6f1c-49c8-8438-d1570abef885)

**Comment by xyz899** [comment]
> ![Screenshot 2024-01-19 114116](https://private-user-images.githubusercontent.com/105791829/298055003-8601d17e-6f1c-49c8-8438-d1570abef885.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU2NzQzMzgsIm5iZiI6MTcwNTY3NDAzOCwicGF0aCI6Ii8xMDU3OTE4MjkvMjk4MDU1MDAzLTg2MDFkMTdlLTZmMWMtNDljOC04NDM4LWQxNTcwYWJlZjg4NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExOVQxNDIwMzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ODdlNDMyNmY0MTZjMWFhY2I3ZmMyY2RlZGM2MjczZWYxNzllNjI0MjRiZjI4YWJlNGZjMTk4YjZkNDZiZWY0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.tiigl3inlWprgdjVHsBwNDqMVYwons9mULTfTMPR_cw)

start by Installing Foundry and its side components :

https://book.getfoundry.sh/getting-started/installation

`curl -L https://foundry.paradigm.xyz | bash`

**PS : Does not work in a Powershell or CMD environment so you need WSL on Windows**
